src/mantarray_desktop_app/mc_comm.py

```diff
@@ -323,6 +332,39 @@ def _handle_incoming_data(self) -> None:
         else:
             raise UnrecognizedSerialCommModuleIdError(module_id)
 
+    def _process_main_module_comm(self, comm_from_instrument: bytes) -> None:
+        packet_body = comm_from_instrument[
+            SERIAL_COMM_ADDITIONAL_BYTES_INDEX:-SERIAL_COMM_CHECKSUM_LENGTH_BYTES
+        ]
+
+        packet_type = comm_from_instrument[SERIAL_COMM_PACKET_TYPE_INDEX]
+        if packet_type == SERIAL_COMM_CHECKSUM_FAILURE_PACKET_TYPE:
+            returned_packet = SERIAL_COMM_MAGIC_WORD_BYTES + packet_body
+            raise SerialCommIncorrectChecksumFromPCError(returned_packet)
+
+        if packet_type == SERIAL_COMM_STATUS_BEACON_PACKET_TYPE:
+            pass  # TODO Tanner (3/17/21): Implement this in a story dedicated to parsing/handling errors codes in status beacons and handshakes
+        elif packet_type == SERIAL_COMM_COMMAND_RESPONSE_PACKET_TYPE:
+            response_data = packet_body[SERIAL_COMM_TIMESTAMP_LENGTH_BYTES:]
+            if len(self._commands_awaiting_response) == 0:
```

Comment:
an empty collections structures are false values in python so you don't need to get the length
```python
if not self._commands_awaiting_respose:
    ...
```